subprojects/file-collections/src/main/java/org/gradle/api/internal/file/collections/DefaultConfigurableFileCollection.java

```diff
@@ -67,6 +69,7 @@ private enum State {
     private boolean disallowChanges;
     private boolean disallowUnsafeRead;
     private ValueCollector value = EMPTY_COLLECTOR;
+    private Spec<Object> taskDependencyFilter = null;
```

Comment:
This subproject uses Java 8, so I think this could and should be a `java.util.function.Predicate`.